functions/java/modules/module1/src/main/java/com/amazonaws/powertools/workshop/Utils.java

```diff
@@ -0,0 +1,183 @@
+                        .key(newObjectKey)
+                        .metadata(metadata)
+                        .build(),
+                RequestBody.fromBytes(processedImage));
+    }
+
+    private static final String UPDATE_FILE_STATUS_MUTATION = "mutation UpdateFileStatus($input: FileStatusUpdateInput) { updateFileStatus(input: $input) { id status transformedFileKey } }";
+    private static final String UPDATE_FILE_STATUS_QUERY = """
+                {
+                    "query": "%s",
+                    "operationName": "UpdateFileStatus",
+                    "variables": {
+                        "input": {
+                            "id": "%s",
+                            "status": "%s"
+                            %s
+                        }
+                    }
+                }""";
+    public static void markFileAs(String fileId, String status, String transformedFileKey) {
+        LOGGER.info("Task in status {}", status);
+        String graphQLMutationResponse = performGraphQLMutation(APPSYNC_ENDPOINT, UPDATE_FILE_STATUS_QUERY.formatted(UPDATE_FILE_STATUS_MUTATION, fileId, status,
+                transformedFileKey != null ? ", \"transformedFileKey\": \"" + transformedFileKey + "\"" : ""));
+        LOGGER.debug(graphQLMutationResponse);
+    }
+
+    private static String performGraphQLMutation(String endpoint, String jsonQuery) {
+        LOGGER.debug(jsonQuery);
+
+        String host;
+        String path;
+        try {
+            host = new URL(endpoint).getHost();
+            path = new URL(endpoint).getPath();
+        } catch (MalformedURLException e) {
+            throw new UpdateFileStatusException(e);
+        }
+
+        SdkHttpFullRequest request = SdkHttpFullRequest
+                .builder()
+                .method(SdkHttpMethod.POST)
+                .putHeader("Host", host)
+                .putHeader("Content-Type", "application/json")
+                .host(host)
+                .encodedPath(path)
+                .protocol("HTTPS")
+                .contentStreamProvider(() -> new ByteArrayInputStream(jsonQuery.getBytes(StandardCharsets.UTF_8)))
+                .build();
+
+        ExecutionAttributes attributes = ExecutionAttributes.builder()
+                .put(AwsSignerExecutionAttribute.AWS_CREDENTIALS, credentialsProvider.resolveCredentials())
+                .put(AwsSignerExecutionAttribute.SERVICE_SIGNING_NAME, "appsync")
+                .put(AwsSignerExecutionAttribute.SIGNING_REGION, AWS_REGION)
+                .build();
+
+        // Perform AWS Signature Version 4 signing
+        Aws4Signer signer = Aws4Signer.create();
+        SdkHttpFullRequest prepRequest = signer.sign(request, attributes);
+
+        // Perform the HTTP call to GraphQL endpoint
+        HttpPost httpPost = new HttpPost(endpoint);
+        for (Map.Entry<String, List<String>> header : prepRequest.headers().entrySet()) {
+            if (header.getKey().equalsIgnoreCase("host")) { continue; }
```

Comment:
@jeromevdl nitpick - I feel like you've discouraged me from using `continue;` before; you could also just invert this expression and put the block below inside

Reply:
thank you but you're nitpicking. It's a not powertools itself ;)

Reply:
I have only one code review mode 🤣  It's up to you to decide what makes sense for the scope. It is _very_ nitpicky.